### PR TITLE
feat: add endpoints for persisting and getting user events

### PR DESCRIPTION
**Pull request by chris-dickson**
This PR adds Hibernate connection properties for persisting values to a database.  We add a simple `EventResource` endpoint that allows for creating of events and listing/search for previous events.  

The `EventType` can be specified as `persistent` meaning that records are stored in the database for the given user.  This is useful for recording things like searches, logins, logouts, etc.  Non-persistent events are logged for metrics and/or dashboarding after the fact (eg/ facet filter, add X to project, etc)

To create an event:
```bash
curl --location --request POST 'http://localhost:3000/api/events' \
--header 'Authorization: Bearer ...
--data-raw '{
    "type": "SEARCH",
    "projectId": 1,
    "value": "covid-19"
}'
```
Results in the following event persisted to the database:
```json
{
    "id": "11d773fb-62a8-44d3-98c9-64efcbd3767f",
    "timestampMillis": 1674058852832,
    "projectId": 1,
    "username": "adam",
    "type": "SEARCH",
    "value": "covid-19"
}
```
A unique id is generated, along with the timestamp of event creation, the projectId, username, type, and value of the event are persisted.

We additionally provide and endpoint for fetching events:
```bash
curl --location --request GET 'http://localhost:3000/api/events?projectId=1&type=SEARCH' \
--header 'Authorization: Bearer ... '
```
Results in:
```json
[
    {
        "id": "a604313e-8a97-4607-987a-aa53c3db8368",
        "timestampMillis": 1674059193970,
        "projectId": 1,
        "username": "adam",
        "type": "SEARCH",
        "value": "covid-19"
    }
]
```

The `projectId` and `type` are required parameters. We inherently filter to only events created by the requesting user. We can optionally provide a `search` term that will filter the results based on the search string being contained in the `value` field of the event.  

eg:
```
```bash
curl --location --request GET 'http://localhost:3000/api/events?projectId=1&type=SEARCH&value=cov' \
--header 'Authorization: Bearer ... '
```

### CODE REVIEW

File: packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/entities/Event.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+import java.io.Serializable;
+import java.time.Instant;
+import java.util.List;
+import java.util.UUID;
+
+@Entity
+@Data
+@Accessors(chain = true)
+@NoArgsConstructor
+public class Event extends PanacheEntityBase implements Serializable {
```
Comment (mwdchang) [inline packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/entities/Event.java:23]: Do we need indices?

Comment (chris-dickson) [reply packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/entities/Event.java:32]: Added

**Review by dvince2 — APPROVED** [review]

**Review by dvince2 — DISMISSED** [review]
LGTM

**Review by tangopapatime — DISMISSED** [review]

**Review by YohannParis — COMMENTED** [review]
Could you add information to the README.md and the [Technical Specifications](https://docs.google.com/document/d/1h_CGQhB0aouIfsOVz79pfuqrgqm7EuBUzaYcekU74aU/edit?usp=sharing)

File: packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/resources/EventResource.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+	}
+
+	/**
+	 * Create an event
+	 * @param event	the {@link Event} instance
+	 * @return			the persisted event instance
+	 */
+	@POST
+	@Transactional
+	public Response putEvent(final Event event) {
```
Comment (dvince2) [inline packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/resources/EventResource.java:57]: Should this be `postEvent` with that annotation?

**Review by dvince2 — COMMENTED** [review]
Unsure if the method name here is intentional or not

File: packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/resources/EventResource.java (hunk 2 above)
Comment (chris-dickson) [reply packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/resources/EventResource.java]: Fixed

**Review by dvince2 — APPROVED** [review]

**Pull request merged**